### DISCUSSION TITLE
The name does not exist in the current context

### DISCUSSION THREAD

**Discussion by foxesknow** [dotnet/roslyn#59274] [Q&A] [answered]
When using a `switch` expression to match types I've noticed something strange. This code:

```
List<string> items = value switch
{
    string i  => new(){i},
    IEnumerable<string> sequence => sequence.ToList(),
    _ => throw new Exception()
};
```

Generates an error `error CS0103: The name 'i' does not exist in the current context`

Changing the `new()` to `new List<string>()` fixes the issue:

```
List<string> items = value switch
{
    string i  => new List<string>(){i},
    IEnumerable<string> sequence => sequence.ToList(),
    _ => throw new Exception()
};
```

Is this a bug?

At first I thought it was the `new()` causing the problem, but this works:
```

List<string> items = value switch
{
    string i  => new(){},
    IEnumerable<string> sequence => sequence.ToList(),
    _ => throw new Exception()
};
```

**Comment by svick** [comment, marked as answer]
This discussion has been converted to the issue https://github.com/dotnet/roslyn/issues/59275, which has been closed as a duplicate of https://github.com/dotnet/roslyn/issues/54193, which has already been fixed.
